### ISSUE TITLE
"find: '>     bin [': No such file or directory" on Windows running update-packages

### ISSUE THREAD

**Issue by DanTup** [flutter/flutter#9117]
I've not used Flutter before but I'm investigating [a bug in Dart Code](https://github.com/Dart-Code/Dart-Code/issues/275) and was asked to clone this repo and run `./bin/flutter update-packages`. When I do this on Windows (10), there are lots of errors like the one in the subject which I presume are unexpected:

```text
M:\Coding\Applications\flutter [master ≡]> ./bin/flutter update-packages
Checking Dart SDK version...
Downloading Dart SDK 1.23.0-dev.10.0...
Unzipping Dart SDK...
Updating flutter tool...
find: '>     bin [': No such file or directory

  ╔════════════════════════════════════════════════════════════════════════════╗
  ║                 Welcome to Flutter! - https://flutter.io                   ║
  ║                                                                            ║
  ║ The Flutter tool anonymously reports feature usage statistics and crash    ║
  ║ reports to Google in order to help Google contribute improvements to       ║
  ║ Flutter over time.                                                         ║
  ║                                                                            ║
  ║ Read about data we send with crash reports:                                ║
  ║ https://github.com/flutter/flutter/wiki/Flutter-CLI-crash-reporting        ║
  ║                                                                            ║
  ║ See Google's privacy policy:                                               ║
  ║ https://www.google.com/intl/en/policies/privacy/                           ║
  ║                                                                            ║
  ║ Use "flutter config --no-analytics" to disable analytics and crash         ║
  ║ reporting.                                                                 ║
  ╚════════════════════════════════════════════════════════════════════════════╝

Downloading Material fonts...
Downloading package sky_engine...
Downloading windows-x64 tools...
Downloading android-arm-profile/windows-x64 tools...
Downloading android-arm-release/windows-x64 tools...
Downloading android-x86 tools...
Downloading android-x64 tools...
Downloading android-arm tools...
Downloading android-arm-profile tools...
Downloading android-arm-release tools...
Running 'flutter packages get' in automated_tests...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in complex_layout...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in microbenchmarks...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in bots...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in devicelab...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in manual_tests...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in tools...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in flutter_gallery...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in flutter_view...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in hello_world...
find: '>     bin [': No such file or directory
Running 'flutter packages get' in layers...
find: '>     bin [': No such file or directory
(snip)...
```

**Comment by abarth** [comment]
@goderbauer

**Comment by goderbauer** [comment]
@DanTup I've seen that error before when running flutter from a third-party shell (like Git Bash or Cygwin or MinGW) on Windows. It's a problem with the underlying Dart infrastructure (see https://github.com/dart-lang/pub/issues/1540). Unfortunately, until that is resolved, Flutter won't be able to support third-party shells on Windows (https://github.com/flutter/flutter/issues/8860).

Are you running Flutter from a third-party shell? Do you see the same error when running from PowerShell or Cmd?

**Comment by DanTup** [comment]
@goderbauer It was just a standard PowerShell shell in Windows 10. I've not seen the issue using Dart or Pub before; though I'm not a heavy Dart user so that might not mean much.

**Comment by goderbauer** [comment]
Interesting... Unfortunately, I cannot reproduce this on my machine. Can you post the output of `flutter doctor` here and can you also tell me which version of PowerShell you are using?

Do you see the same issue when running `flutter update-packages` in the regular Cmd prompt?

**Comment by DanTup** [comment]
Actually, I was wrong about the shell! :( I just tried it again and got `Error: Unable to find git in your PATH.`. That reminded me that when I did that before I just launched PoSh from within GitHub Desktop (which sets the path to include the version of Git it shipped with). So, it seems to be caused by using GitHub for Windows shell (which is likely why I never say it with Pub; I only ever launch PoSh this way for Git stuff).

So, this is certainly a dupe of the other issue and I've confirmed it does indeed work from a "standard" PoSh shell (once Git is available in the path).

Apologies for the misleading info; probably makes sense to close this as a dupe of the Dart case.
